lib/buffer.js

```diff
@@ -672,23 +672,27 @@ Buffer.prototype.fill = function fill(val, start, end, encoding) {
       encoding = end;
       end = this.length;
     }
-    if (val.length === 1) {
-      var code = val.charCodeAt(0);
-      if (code < 256)
-        val = code;
-    }
-    if (val.length === 0) {
-      // Previously, if val === '', the Buffer would not fill,
-      // which is rather surprising.
-      val = 0;
-    }
+
     if (encoding !== undefined && typeof encoding !== 'string') {
       throw new TypeError('encoding must be a string');
     }
-    if (typeof encoding === 'string' && !Buffer.isEncoding(encoding)) {
+    var normalizedEncoding = internalUtil.normalizeEncoding(encoding);
+    if (normalizedEncoding === undefined) {
       throw new TypeError('Unknown encoding: ' + encoding);
     }
 
+    if (val.length === 0) {
+      // Previously, if val === '', the Buffer would not fill,
+      // which is rather surprising.
+      val = 0;
+    } else if (val.length === 1) {
+      var code = val.charCodeAt(0);
+      if ((normalizedEncoding === 'utf8' && code < 128) ||
+          normalizedEncoding === 'latin1') {
```

Comment:
@trevnorris I switched the blocks around here so that this is after `normalizeEncoding`, so there’s no need to worry about alternate names